### ISSUE TITLE
Implement/refactor GA tracking events for help menu and URL search

### ISSUE THREAD

**Issue by felixarntz** [google/site-kit-wp#4057]
## Feature Description

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* 

## Implementation Brief

* 

### Test Coverage

* 

## QA Brief

* 

## Changelog entry

*

**Comment by eugene-manuilov** [comment]
IB :heavy_check_mark:

**Comment by eugene-manuilov** [comment]
> Note that these events should be present in both the old and new header bar (see Implement new Unified Dashboard header design/UI behind feature flag #4048).

@johnPhillips IB almost looks good to me, but AC says that we also need to add the same events to the new header bar. I am going to assign this ticket back to you. You will probably need to wait till IB for #4048 is written to be able to add the remaining part.

**Comment by johnPhillips** [comment]
@eugene-manuilov I just spoke to @tofumatt and my understanding is that the final AC here is more of an FYI, as the header will likely use the same existing URL search component. I don't think it requires anything in the IB.

**Comment by eugene-manuilov** [comment]
Ok, then IB :heavy_check_mark:

**Comment by kuasha420** [comment]
QA: :heavy_multiplication_x: 

This did not quite pass. See the following. 

* :white_check_mark:   `click_outgoing_links` events are being fired with the correct `{viewContext}_headerbar_helpmenu` event category with the `viewContext` being correctly inferred as `dashboard`. `module` and `pageDasboard` accordingly. 
* :x: `open_urldetails` is firing with just the `dashboard` as category instead of `dashboard_urlsearch-widget`.
    * ![Screenshot_20211005_052832](https://user-images.githubusercontent.com/6187401/135938025-7b9ef3dd-7b7a-400f-bd8a-d8ba17190c7e.png)
* :grey_question: How to get The help menu feature tour? I have tried setting the initial plugin version back but it didn't bring up anything. A note about that in QAB would be appreciated. 

Also, should this ticket wait for #4048? @felixarntz 

Also, as John is not available and he's not available, I'm not sure whom to assign for the execution. Leaving as unassigned to Execution for now. 

cc @eugene-manuilov

**Comment by eugene-manuilov** [comment]
Good catch, @kuasha420. I have created a new pull request that fixes the GA event category issue. @aaemnnosttv do you mind reviewing #4188?

>  How to get The help menu feature tour? I have tried setting the initial plugin version back but it didn't bring up anything.

@kuasha420 try to delete dismissed feature tours for your user.

> Also, should this ticket wait for #4048?

I'll update that ticket to reflect the required changes in the URL search component.

**Comment by aaemnnosttv** [comment]
@eugene-manuilov the follow-up PR has been merged 👍 

@kuasha420 would you please give this another pass on `main`?

**Comment by kuasha420** [comment]
QA: :heavy_check_mark: 

Tested on `main`. LGTM!

* :white_check_mark:   `click_outgoing_links` events are being fired with the correct `{viewContext}_headerbar_helpmenu` event category with the `viewContext` being correctly inferred as `dashboard`. `module` and `pageDasboard` accordingly. 
* :white_check_mark: `open_urldetails` is firing with the `dashboard_urlsearch-widget` as category instead of `dashboard_urlsearch-widget`.
    * ![Screenshot_20211006_002321](https://user-images.githubusercontent.com/6187401/136080895-0a9fbfe1-4064-425f-b58b-b35fbd4031f6.png)
* :white_check_mark: Feature Tour tooltip is also using the new Event category. 
    * ![Screenshot_20211006_002022](https://user-images.githubusercontent.com/6187401/136081071-2ffa04f0-95a1-49c5-a0a1-cc30aeea6e18.png)
